chore(create-expo): move package from `expo/expo-cli` to `expo/expo`

byCedric opened the pull request:
# Why

Moving this actively used/developed package from [expo/expo-cli](https://github.com/expo/expo-cli) to [expo/expo](https://github.com/expo/expo).

> ⚠️ ~~Blocked by `@expo/package-manager` changes being merged only in `sdk-49` and not `main`. See https://github.com/expo/expo/pull/24376~~ 
> Merged in this PR!

# How

- Copied project into **packages/create-expo-app**
- Applied various changes to fit it in the repository

# Test Plan

See CI

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

byCedric committed:
chore(create-expo-app): move project from `expo/expo-cli`

byCedric committed:
chore(create-expo-app): rename project to `create-expo-app`

byCedric committed:
chore(create-expo-app): update repository in `package.json`

byCedric committed:
chore(create-expo-app): drop publish config from `package.json`

byCedric committed:
refactor(create-expo-app): replace implicit dependency `rimraf` with `git clean`

byCedric committed:
fix(create-expo-app): use proper base `tsconfig.json`

byCedric committed:
fix(create-expo-app): remove test runner option from jest

byCedric committed:
fix(create-expo-app): add missing `ncc`

byCedric committed:
chore(create-expo-app): exclude `build` folder from repository

byCedric committed:
fix(create-expo-app): exclude `build` from linting

byCedric committed:
fix(create-expo-app): resolve all linting issues

byCedric committed:
chore: update yarn lockfile

byCedric committed:
refactor(create-expo-app): use `expo-module-script`

byCedric committed:
chore: update yarn lockfile again

byCedric committed:
refactor(create-expo-app): drop implicit dependency `cross-env`

byCedric committed:
docs(create-expo-app): add missing changelog from manually constructed history

byCedric commented:
Lmao, bad bot @expo-bot

byCedric committed:
fix(package-manager): sync version of `@expo/package-manager` from `sdk-49`

byCedric committed:
fix(cli): update `@expo/cli` to also use version from `@expo/package-manager`

EvanBacon commented on packages/create-expo-app/jest.config.js:
```diff
@@ -0,0 +1,18 @@
+const path = require('path');
+
+const roots = ['./src'];
```
I'd recommend splitting this into two different scripts like we do with expo/cli and have one set of testing be part of the check packages, and the other be part of an e2e test workflow like we do with CLI and fingerprint.

EvanBacon commented on packages/create-expo-app/package.json:
```diff
@@ -0,0 +1,60 @@
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/expo/expo.git",
+    "directory": "packages/create-expo-app"
```
Make the folder be `create-expo` and move stuff accordingly. `bun create expo` > `bun create expo-app`.

byCedric committed:
refactor(create-expo): rename to `create-expo`

byCedric committed:
refactor(create-expo): split up jest normal and e2e tests

byCedric committed:
refactor(create-expo): convert e2e test to typescript

byCedric committed:
chore: add `create-expo` e2e tests to cli workflow

byCedric committed:
fix: `minipass` and `tar` issues in both `@expo/cli` and `create-expo` (#24402)

# Why

This upgrades both `create-expo` and `@expo/cli` to the latest
`minipass@3.x.x`. There are newer versions available, but want to
minimize potential breaking issues introduced by this.

# How

- Upgraded to `minipass@3.3.6` to remove `@types/minipass`
(`minipass@3.3.6` ships its own types)

# Test Plan

See if CI passes.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

byCedric committed:
docs: add changelog entries about `tar` and `minipass`

byCedric committed:
chore(package-manager): rebuild definition files
